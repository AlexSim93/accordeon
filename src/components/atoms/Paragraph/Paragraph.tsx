import styled from 'styled-components';
import { color, typography, space } from 'styled-system';

export const Paragraph = styled.p`
  ${space}
  ${typography}
  ${color}
`;

Paragraph.displayName = 'Paragraph';
