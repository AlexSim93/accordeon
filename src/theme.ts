const breakpoints = ['768px', '1440px'];

export default {
  breakpoints,
  fontSizes: {
    xxs: '10px',
    xs: '12px',
    s: '14px',
    m: '16px',
    ml: '20px',
    l: '24px',
    xl: '32px',
    xxl: '40px',
    xxxl: '56px',
  },
  space: { // s 8, m 16, l 24
    xxxxs: '4px',
    xxxs: '8px',
    xxs: '12px',
    xs: '16px',
    s: '24px',
    sm: '32px',
    m: '40px',
    ml: '48px',
    l: '56px',
    xl: '64px',
    xxl: '72px',
    xxxl: '140px',
    xxxxl: '200px',
  },
  mediaQueries: {
    small: `@media screen and (min-width: ${breakpoints[0]})`,
    medium: `@media screen and (min-width: ${breakpoints[1]})`,
  },
  colors: {
    white: '#FFFFFF', // title text
    blackPearlLighten48: 'rgba(8, 13, 38, 0.48)', // border
    medianGrey: '#A7A7C8', // description text
    darkGreyLighten16: 'rgba(103, 107, 139, 0.16)', // accordion bg
    neutralDarkGrey: '#C4C4C4', // icon
    darkBlue: '#080D26', // page bg
    dodgerBlue: '#564FFF', // button bg
  },
  fonts: {
    Montserrat: '"Montserrat", sans-serif',
  },
  radii: {
    s: '4px',
  },
};
