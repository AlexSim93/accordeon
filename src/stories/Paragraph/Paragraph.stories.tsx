import React from 'react';
import { storiesOf } from '@storybook/react';
import { Box, Flex } from 'reflexbox';
import { ThemeProvider } from 'styled-components';
import { Paragraph } from '../../components/atoms/Paragraph';
import notes from './readme.md';
import { Text } from '../../components/atoms/Text';
import theme from '../../theme';

storiesOf('Paragraph', module).add(
  'Paragraph',
  () => (
    <ThemeProvider theme={theme}>
      <Flex flexDirection="column">
        <Text color="black" fontWeight="600" fontSize="xl">
          Paragraph
        </Text>
        <Box mb={24}>
          <Text color="black" fontSize="l">
            Default paragraph
          </Text>
          <Paragraph>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corporis
            consectetur mollitia dicta voluptas soluta! Accusantium, doloremque
            neque, deleniti temporibus eius corporis veritatis doloribus quam
            dolorum, impeditFaperiam? Corrupti, animi eos. Lorem ipsum dolor sit
            amet, consectetur adipisicing elit. Corporis consectetur mollitia
            dicta voluptas soluta! Accusantium, doloremque neque, deleniti
            temporibus eius corporis veritatis doloribus quam dolorum,
            impeditFaperiam? Corrupti, animi eos. Lorem ipsum dolor sit amet,
            consectetur adipisicing elit. Corporis consectetur mollitia dicta
            voluptas soluta! Accusantium, doloremque neque, deleniti temporibus
            eius corporis veritatis doloribus quam dolorum, impeditFaperiam?
            Corrupti, animi eos. Lorem ipsum dolor sit amet, consectetur
            adipisicing elit. Corporis consectetur mollitia dicta voluptas
            soluta! Accusantium, doloremque neque, deleniti temporibus eius
            corporis veritatis doloribus quam dolorum, impeditFaperiam?
            Corrupti, animi eos.
          </Paragraph>
        </Box>
        <Box mb={24}>
          <Text color="black" fontSize="l">
            Paragraph fontSize large, color blackPearlLighten48
          </Text>
          <Paragraph color="blackPearlLighten48" fontSize="l">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Corporis
            consectetur mollitia dicta voluptas soluta! Accusantium, doloremque
            neque, deleniti temporibus eius corporis veritatis doloribus quam
            dolorum, impedit aperiam? Corrupti, animi eos.Lorem ipsum dolor sit
            amet, consectetur adipisicing elit. Corporis consectetur mollitia
            dicta voluptas soluta! Accusantium, doloremque neque, deleniti
            temporibus eius corporis veritatis doloribus quam dolorum,
            impeditFaperiam? Corrupti, animi eos.Lorem ipsum dolor sit amet,
            consectetur adipisicing elit. Corporis consectetur mollitia dicta
            voluptas soluta! Accusantium, doloremque neque, deleniti temporibus
            eius corporis veritatis doloribus quam dolorum, impeditFaperiam?
            Corrupti, animi eos.Lorem ipsum dolor sit amet, consectetur
            adipisicing elit. Corporis consectetur mollitia dicta voluptas
            soluta! Accusantium, doloremque neque, deleniti temporibus eius
            corporis veritatis doloribus quam dolorum, impeditFaperiam?
            Corrupti, animi eos.
          </Paragraph>
        </Box>
      </Flex>
    </ThemeProvider>
  ),
  { notes },
);
