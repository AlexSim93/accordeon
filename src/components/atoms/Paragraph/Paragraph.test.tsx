import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { Paragraph } from './Paragraph';
import 'jest-styled-components';
import theme from '../../../theme';

describe('Paragraph renders correctly', () => {
  test('Paragraph renders correctly with color=white and fontFamily=Montserrat', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Paragraph color="white" fontFamily="Montserrat" />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
