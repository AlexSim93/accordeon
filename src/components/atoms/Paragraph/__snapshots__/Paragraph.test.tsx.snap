// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders correctly Paragraph renders correctly with color=white and fontFamily=Montserrat 1`] = `
.c0 {
  font-family: "Montserrat",sans-serif;
  color: #FFFFFF;
}

<p
  className="c0"
  color="white"
  fontFamily="Montserrat"
/>
`;
